sw/host/tests/rom/e2e_bootstrap_entry/src/main.rs

```diff
@@ -570,11 +570,56 @@ fn test_bootstrap_phase2_read(opts: &Opts, transport: &TransportWrapper) -> Resu
     Ok(())
 }
 
+fn test_bootstrap_watchdog_check(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    
+    let uart = transport.uart("0")?;
+    let mut console = UartConsole {
+        timeout: Some(opts.timeout),
+        exit_success: Some(Regex::new(r"bootstrap:1\r\n")?),
+        exit_failure: Some(Regex::new(r"BFV:")?),
+        ..Default::default()
+    };
+    // Apply bootstrap pin strapping.
+    let _bs = BootstrapTest::start(transport, opts.init.bootstrap.options.reset_delay)?;
+    transport.apply_pin_strapping("ROM_BOOTSTRAP")?;
+    // Reset the chip.
+    transport.reset_target(opts.init.bootstrap.options.reset_delay, true)?;
+    
+    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
+    if result != ExitStatus::ExitSuccess {
+        bail!("FAIL: {:?}", result);
+    };
+    // Verify that the chip responds to `READ_STATUS` (`0x05`) with `0x00`.
+    let spi = transport.spi("0")?;
+    assert_eq!(SpiFlash::read_status(&*spi)?, 0x00);
+    
+    // Discard buffered messages before interacting with the console.
+    uart.clear_rx_buffer()?;
+    console.buffer.clear();
+    
+    // Release bootstrap pin strapping and wait for 2 seconds.
+    transport.remove_pin_strapping("ROM_BOOTSTRAP")?;
+    std::thread::sleep(std::time::Duration::from_secs(2));
+
+    //Verify that the chip responds to `READ_STATUS` (`0x05`) with `0x00`.
+    assert_eq!(SpiFlash::read_status(&*spi)?, 0x00);
+
+    // Verify that there was no output from UART.
+    console.exit_success = Some(Regex::new(r".+")?);
+    console.exit_failure = Some(Regex::new(r".+")?);
+    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
+    if result != ExitStatus::Timeout {
+        bail!("FAIL: {:?}", result);
+    };
+
+    Ok(())
```

Comment:
I think we want something like the following. Please review and verify. Thanks!

```suggestion
    let _bs = BootstrapTest::start(transport, opts.init.bootstrap.options.reset_delay)?;
    let spi = transport.spi("0")?;
    let uart = transport.uart("0")?;
    let mut console = UartConsole {
        timeout: Some(Duration::new(2, 0)),
        exit_success: Some(Regex::new(r".+")?),
        exit_failure: Some(Regex::new(r".+")?),
        ..Default::default()
    };
    
    // Verify that the chip responds to `READ_SFDP` (`0x5a`). 
    let sfdp_ = SpiFlash::read_sfdp(&*spi)?;
    // Release bootstrap pin strapping and verify that we don't receive
    // anything over UART until the console times out.
    uart.clear_rx_buffer()?;
    transport.remove_pin_strapping("ROM_BOOTSTRAP")?;
    let result = console.interact(&*uart, None, Some(&mut std::io::stdout()))?;
    if result != ExitStatus::Timeout {
        bail!("FAIL: {:?}", result);
    };
    // Verify that the chip is still in bootstrap.
    let sfdp_ = SpiFlash::read_sfdp(&*spi)?;
    
    Ok(())
```